src/helpers/SpotLightHelper.js

```diff
@@ -67,6 +64,24 @@ class SpotLightHelper extends Object3D {
 		this.light.updateWorldMatrix( true, false );
 		this.light.target.updateWorldMatrix( true, false );
 
+		// update the local matrix based on the parent and light target transforms
+		if ( this.parent ) {
+
+			this.matrix
+				.copy( this.parent.matrixWorld )
```

Comment:
Is this matrix necessarily current?

Reply:
I suppose that's true - I can call `this.parent.updateMatrixWorld` but it has the same caveats as mentioned in https://github.com/mrdoob/three.js/pull/27487#discussion_r1452110938 in that it will undermine the users `parent.matrixWorldAutoUpdate` setting if it's set to false.